File: JTalk/src/Models/Address.java
package Models;

import com.thoughtworks.xstream.annotations.XStreamAlias;

@XStreamAlias("address")
public class Address {
    public Address() {
    }

    public Address(int Priority, String AddressText) {
        this.Priority = Priority;
        this.AddressText = AddressText;
    }

    @XStreamAlias("priority")
    public Integer Priority;

    @XStreamAlias("addressText")
    public String AddressText;
}

File: TalkSharp/TalkSharp/Models/Address.cs
using System.Xml.Serialization;

namespace TalkSharp.Models
{
    [XmlRoot("address")]
    public class Address
    {
        // Xml Serializer insists on having a parameterless constructor
        public Address()
        {
        }

        public Address(int Priority, string AddressText)
        {
            this.Priority = Priority;
            this.AddressText = AddressText;
        }

        [XmlElement("priority")]
        public int Priority { get; set; }

        [XmlElement("addressText")]
        public string AddressText { get; set; }
    }
}